### ISSUE TITLE
The question about model training

### ISSUE THREAD

**Issue by HH-yx** [liruiw/GA-DDPG#8]
Dear Liruiw:
I have a question about model training.
I would like to know the idea of model training is to train offline first, using behavioral clones to learn the strategy of omg planning, and then train online, using the trained model to interact with the environment and fine-tune the strategy with hindsight goals

**Comment by liruiw** [comment]
Thanks for the questions. The first part does not have to offline only (although you could as in normal BC). The idea is to let the policy jointly learn from experts (through BC loss) and from interactions (through value function loss). The hindsight goals apply to cases where we don't have ground truth goals (for instance no perfect state estimates in the real world and therefore cannot run planners). In these cases, we can self-supervise goals by using hindsights.
